lessons/beginners/list/index.md

```diff
@@ -30,204 +20,253 @@ print(cisla)
-že se proměnná „mění aniž na ni sáhneme”:
+že se „mění proměnná, aniž na ni sáhneš”:
 
 ```python
-a = [1, 2, 3]   # vytvoření seznamu
-b = a           # tady se nový seznam nevytváří
+a = [1, 2, 3]   # Vytvoření seznamu
+b = a           # Tady se nový seznam nevytváří!
 
 # seznam vytvořený v prvním řádku má teď dvě jména: "a" a "b",
-# ale stále pracujeme jenom s jedním seznamem
+# ale stále pracuješ jenom s jedním seznamem
 
 print(b)
 a.append(4)
 print(b)
 ```
 
+
 ## Další způsoby, jak měnit seznamy
 
-Kromě metody `append`, která přidává
-jediný prvek, existuje metoda `extend`,
-která umí přidávat prvků víc.
-Prvky k přidání jí předáme ve formě seznamu:
+Kromě metody `append`, která přidává jediný prvek na konec, existuje
+spousta dalších metod, které seznamy mění.
+Všechny udělají změny přímo v daném seznamu a (kromě `pop`) vrací `None`:
+
+* `extend()` přidá více prvků najednou,
+* `insert()` přidá prvek na danou pozici,
+* `pop()` odebere *a vrátí* poslední prvek,
+* `remove()` odstraní první výskyt daného prvku,
+* `sort()` seznam seřadí (řetězce podle “abecedy”, čísla vzestupně),
+* `reverse()` obrátí pořadí prvků,
+* `clear()` odstraní všechny prvky.
+
+{{ figure(img=static('methods.svg'), alt="Tahák") }}
+
+Například:
 
 ```python
-dalsi_prvocisla = [23, 29, 31]
-prvocisla.extend(dalsi_prvocisla)
-print(prvocisla)
+zviratka = ['pes', 'kočka', 'králík']
+zviratka.append('morče')      # ['pes', 'kočka', 'králík', 'morče']
+zviratka.insert(2, 'had')     # ['pes', 'kočka', 'had', 'králík', 'morče']
+zviratka.pop()                # ['pes', 'kočka', 'had', 'králík'], vrátí 'morče'
+zviratka.remove('had')        # ['pes', 'kočka', 'králík']
+zviratka.sort()               # ['kočka', 'králík', 'pes']
+zviratka.reverse()            # ['pes', 'králík', 'kočka']
+zviratka.clear()              # []
 ```
 
-Metoda `extend` umí pracovat i s jinými
-typy než se seznamy – ráda zpracuje cokoli, přes
-co umí cyklit `for`: např.
-jednotlivé znaky řetězců, řádky souborů, nebo čísla z `range()`.
+## Vybírání ze seznamů
+
+Často budeš ze seznamu chtít vybrat prvek na určité pozici.
+To funguje jako u řetězců: do hranatých závorek dáš číslo prvku.
+Stejně jako u řetězců se čísluje od nuly a záporná čísla číslují od konce.
 
 ```python
-seznam = []
-seznam.extend('abcdef')
-seznam.extend(range(10))
-print(seznam)
+zviratka = ['pes', 'kočka', 'králík']
+print(zviratka[2])
 ```
 
-## Měnění prvků
+Hranatými závorkami můžeš získat i podseznam.
+[Diagram z materiálů k řetězcům]({{ lesson_url('beginners/str-index-slice')}}#slicing-diagram)
+ukazuje, jak u takového „sekání” číslovat:
+funguje to stejně, jen místo menšího řetězce dostaneš menší seznam.
+
+```python
+print(zviratka[2:-3])
+```
 
-Ale dost přidávání.
-Seznamům se dají i měnit jednotlivé prvky
-a to jednoduše tak, že do prvku přiřadíme,
-jako by to byla proměnná:
+„Sekáním“ vzniká nový seznam – když pak ten původní změníš, v podseznamu se
+to neprojeví.
```

Comment:
I don't think, I fully understand this sentence. Is it meant that if you put the sliced list in a variable, and change the original list, then the sublist (is that the official term? The English equivalent to the word *podseznam* isn't written in the materials.) doesn't change? Or the sentence says something else?

Reply:
Yes, that's what I meant to say.
Would this be better?
````suggestion
„Sekáním“ vzniká nový seznam – když pak ten původní změníš, v novém menším seznamu se
to neprojeví.
````

*sublist* is right.